app/_src/production/dp-config/dpp-on-kubernetes.md

```diff
@@ -180,6 +180,18 @@ On Kubernetes, `Dataplane` resource is automatically created by kuma-cp. For eac
 
 To join the mesh in a graceful way, we need to first make sure the application is ready to serve traffic before it can be considered a valid traffic destination.
 
+{% if_version lte:2.3.x %}
+
+If any of the init containers experience network connectivity issues we suggest trying setting `runtime.kubernetes.injector.sidecarContainer.waitForDataplaneReady` to `true` so that the init container for the sidecar finishes only when the sidecar is ready to serve traffic.
```

Comment:
There's also an annotation that can be added on a per pod basis. We should make sure it's documented here and in   the annotations page.

Reply:
Maybe it's worth explaining that by default all containers start in parallel and you have no guarantee that the application won't start sending requests before the sidecar is ready. This would make these requests fail.

If this is required you can use the configuration ...

WDYT? Explain the problem before the fix

Reply:
> all containers start in parallel

they don't - https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#differences-from-regular-containers

> If you specify multiple init containers for a Pod, kubelet runs each init container sequentially. 

they run sequentially, that's why the thing that Lukasz implemented works.

Reply:
The doc you linked to are for init-containers right?
it might be that they don't start in parallel but they don't wait for the previous one to be ready before starting the next one...